os/kernel/binary_manager/binary_manager_load.c

```diff
@@ -377,6 +377,21 @@ static int binary_manager_reload(char *bin_name)
 		return BINMGR_NOT_FOUND;
 	}
 
+	if (reload_type == RELOAD_UPDATE) {
+		int version;
+		binary_header_t header_data;
+
+		ret = binary_manager_read_header(bin_idx, BIN_USEIDX(bin_idx) ^ 1, &header_data);
```

Comment:
I think set/get version is very important for yours, it had better make internal function to reuse it in future or for now.